pkg/compose/hash.go

```diff
@@ -40,3 +41,23 @@ func ServiceHash(o types.ServiceConfig) (string, error) {
 	}
 	return digest.SHA256.FromBytes(bytes).Encoded(), nil
 }
+
+// From a top-level Volume Configuration, creates a unique hash ignoring
+// DriverOpts,  External and Extensions
+func VolumeHash(o types.VolumeConfig) (string, error) {
+	// We can't have these attributes into account, because when converting
+	// from a volume inspect there is no way to deduce these attributes back
+	// to a VolumeConfig
+	if o.Driver == "" { // (TODO: jhrotko) This probably should be fixed in compose-go
+		o.Driver = "local"
+	}
+	o.External = false // the name can change. Need to think about this case
```

Comment:
if volume is external you should just never enter this func

Reply:
there is an edge case where you can change the `name` to of an external volume and we need to update the service. The current code does not comtemplate this case. I need to implement this

Reply:
by nature, external resources are not managed by compose. There's nothing we can and should do to cover this scenario

Reply:
I better understand your point, the issue you refer to is a special case for a feature we don't support (yet): recreate services when referred resource is reconfigured. I agree we could kill 2 birds with one stone here, but this makes many changes at once.
Maybe split this into 2 separate PRs ?
- 1 to recreate volumes on config changes
- 1 to detect named resource changed (ensureService to inspect mounted volumes for container and check name matches expectation)

Reply:
I applied both solutions to this PR

Reply:
sure, I mean it could be simpler to review if we split this effort into two separate PRs